sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -45,7 +45,12 @@ public BufferHolder(UnsafeRow row) {
   }
 
   public BufferHolder(UnsafeRow row, int initialSize) {
-    this.fixedSize = UnsafeRow.calculateBitSetWidthInBytes(row.numFields()) + 8 * row.numFields();
+    int bitsetWidthInBytes = UnsafeRow.calculateBitSetWidthInBytes(row.numFields());
+    if (row.numFields() > (Integer.MAX_VALUE - initialSize) / 8) {
+      throw new UnsupportedOperationException(
+        "Cannot create BufferHolder from input UnsafeRow because it is too big.");
```

Comment:
`...too big` might be a bit to vague.... Can you use something like `...exceeds the maximum number of variables (268435455).`